### ISSUE TITLE
serving static files

### ISSUE THREAD

**Issue by jxs** [ratpack/ratpack#981]
Hi there, I am trying to serve static files, but documentation seems outdated, as it suggests to call [files("assets")](https://github.com/ratpack/ratpack/blob/master/ratpack-core/src/main/java/ratpack/handling/Chain.java#L311) but there doesn't seem to be a method ***find*** for String with location.
Even if call ***files()*** with 0 arguments, i get ***ratpack.file.BaseDirRequiredException: no base dir set for application***
I can submit a PR with the updates, but would like to clarify the simplest way to serve statics with the default ***Base Dir*** set

**Comment by danhyun** [comment]
You cannot serve files without declaring some kind of base directory.

You'll need to make use of [`ServerConfigBuilder#baseDir`](https://ratpack.io/manual/current/api/ratpack/server/ServerConfigBuilder.html#baseDir-java.io.File-)

As a quick Java example:

```java
RatpackServer.start(serverSpec -> serverSpec.
  serverConfig(configBuilder -> configBuilder.baseDir(BaseDir.find()))
);
```

This is making use of [`BaseDir.find`](https://ratpack.io/manual/current/api/ratpack/server/BaseDir.html#find--) which will look for a default marker file `.ratpack` to determine as the base dir of the application. All files are resolved relatively against this directory.

**Comment by danveloper** [comment]
Yeah you're right the manual docs are incorrect for the `files()` chain action. However, [this section](https://ratpack.io/manual/current/launching.html#base_dir) on setting up the base dir is correct. So a simple Java app might look something like:

```java
public class Main {
  public static void main(String[] args) throws Exception {
    RatpackServer.start(spec ->
      .serverConfig(d -> d.baseDir(BaseDir.find()))
      .handlers(chain ->
        .files(f -> f.dir("public").indexFiles("index.html"))
      )
    )
  }
}
```

And this is given a project structure of:

```
.
├── build.gradle
└── src
    └── main
        ├── java
        │   └── tld
        │       └── company
        │           └── app
        │               └── Main.java
        └── resources
            ├── .ratpack
            └── public
                └── index.html
```

We need to update the docs to properly reflect this.

**Comment by jxs** [comment]
ok, thanks, I will submit a PR with the updates

**Comment by alkemist** [comment]
Resolved by #982 & 39790b081d6084890d157845ff23fa9f9aa95b4a.
